src/main/java/org/fundacionjala/coding/danielmontecinos/bank_ocr/BankOcr.java

```diff
@@ -0,0 +1,210 @@
+                        + " _|");
+    }
+
+    /**
+     * Constructor private.
+     */
+    private BankOcr() {
+
+    }
+
+    /**
+     * @param value used to obtain its int representation
+     * @return "?" if the value is not in range 0-9 otherwise
+     * return a number between 0 and 9.
+     */
+    private static String getKey(final String value) {
+        String key = UNKNOWN_NUMBER;
+        for (Map.Entry<Integer, String> entry : STRING_MAP.entrySet()) {
+            if (entry.getValue().equals(value)) {
+                key = entry.getKey().toString();
+            }
+        }
+        return key;
+    }
+
+    /**
+     * This method determines if an scanned account is legible.
+     *
+     * @param account image of type String.
+     * @return true or false
+     */
+    private static boolean isLegible(final String account) {
+        boolean isCorrect = true;
+
+        for (int i = 0; i < account.length(); i++) {
+            if (!Character.isDigit(account.charAt(i))) {
+                isCorrect = false;
+                break;
+            }
+        }
+        return isCorrect;
+    }
+
+    /**
+     * This method validate if an account is correct or not.
+     *
+     * @param account String variable is the account to validate
+     * @return true if the passed account is true, false otherwise.
+     */
+    static boolean validateAccount(final String account) {
+        if (!(account.length() == CORRECT_ACCOUNT_LENGTH) || !isLegible(account)) {
+            return false;
+        }
+
+        int checksum = 0;
+        int factor = MULTIPLY_FACTOR;
+
+        for (String value : account.split(BY_EACH_CHARACTER)) {
+            checksum += Integer.parseInt(value) * factor;
+            factor--;
+        }
+
+        return checksum % MODULUS_FACTOR == 0;
+
+    }
+
+    /**
+     * This method is in charge to give one of three statues to the account".
+     * "", "ERR" or "ILL"
+     *
+     * @param account String variable to determine its status.
+     * @return an String result with the values mentioned earlier.
+     */
+    static String getAccountStatus(final String account) {
+        return !isLegible(account) ? ILLEGIBLE : (!validateAccount(account)) ? ERROR : EMPTY_STRING;
+    }
+
+    /**
+     * This method returns the string representation of the scanned image.
+     *
+     * @param scannedImage array string.
+     * @return String representation of scanned image.
+     */
+    static String accountRepresentation(final String[] scannedImage) {
+        return Stream.of(scannedImage)
+                .reduce(BY_EACH_CHARACTER, (a, b) -> a + getKey(b));
+    }
+
+    /**
+     * This method take and scanned account of exactly 9 digits and return
+     * an array representation of them.
+     *
+     * @param scannedAccount String, represents the account.
+     * @return an array of nine string digits.
+     */
+    static String[] parseScannedNumbers(final String scannedAccount) {
```

Comment:
Carlos I don't know how to improve this method, I've tried to use a List instead an Array but I only got incorrect result, I'm really stuck with this method